src/main/java/move/CannonMovement.java

```diff
@@ -0,0 +1,67 @@
+package move;
+
+import direction.Direction;
+import direction.Point;
+import java.util.ArrayList;
+import java.util.List;
+import piece.Pieces;
+
+public class CannonMovement implements MovementRule {
+    private static final String CANNON_EXPRESSION = "n";
+
+    @Override
+    public void validateDestination(Point from, Point to) {
+        validateStraightDestination(from, to);
+        validateNotSamePosition(from, to);
+    }
+
+    @Override
+    public void checkPaths(Pieces allPieces, Point from, Point to) {
+        if(calculateCannonPieceCountInPaths(allPieces, from, to) >= 1) {
+            throw new IllegalArgumentException("[ERROR] 포가 존재하여 움직일 수 없습니다.");
+        }
+        if (calculateNotCannonPieceCountInPaths(allPieces, from, to) != 1) {
+            throw new IllegalArgumentException("[ERROR] 포는 포를 제외한 하나의 기물을 넘어야 합니다.");
+        }
+    }
```

Comment:
사소하지만 종종 컨벤션이 맞지 않는 부분이 보이는 것 같아요.

```suggestion
    @Override
    public void checkPaths(Pieces allPieces, Point from, Point to) {
        if (calculateCannonPieceCountInPaths(allPieces, from, to) >= 1) {
            throw new IllegalArgumentException("[ERROR] 포가 존재하여 움직일 수 없습니다.");
        }
        if (calculateNotCannonPieceCountInPaths(allPieces, from, to) != 1) {
            throw new IllegalArgumentException("[ERROR] 포는 포를 제외한 하나의 기물을 넘어야 합니다.");
        }
    }
```

추후 미미가 여러 사람과 협업을 하게 된다면 이러한 부분이 잘 지켜졌을 때 추가적인 커뮤니케이션 비용 없이 코드의 일관성을 유지할 수 있지 않을까 의견 드립니다. 전반적으로 확인 한 번 부탁드려요~

Reply:
의식적으로 신경쓰는 연습을 해야 할 것 같습니다..! 피드백 감사합니다 👍🏻